File: backend/src/models/IntegrationSession.js
import mongoose from 'mongoose'
import createSchema from './utils/createSchema'

const IntegrationSessionSchema = createSchema({
  userId: {type: String, required: true, index: true},
  alias: {type: String, required: true},
  protocol: {type: String, required: true, enum: ['rpc', 'mcp']},
  lastSessionId: {type: String},
})

IntegrationSessionSchema.index({userId: 1, alias: 1, protocol: 1}, {unique: true})

const IntegrationSession = mongoose.model('IntegrationSession', IntegrationSessionSchema)

export default IntegrationSession

File: backend-v2/internal/models/integration_session.go
package models

type IntegrationSession struct {
	UserID        string `json:"userId" bson:"userId"`
	Alias         string `json:"alias" bson:"alias"`
	Protocol      string `json:"protocol" bson:"protocol"`
	LastSessionId string `json:"lastSessionId" bson:"lastSessionId"`
}
